### ISSUE TITLE
cryptography is not installed on MacOS

### ISSUE THREAD

**Issue by pierreguillot** [pure-data/deken#172]
When I install deken on MacOS 10.12.6 I got 
```
Using cached cryptography-2.1.3.tar.gz
Complete output from command python setup.py egg_info:
error in cryptography setup command: Invalid environment marker: python_version < '3'
    
    ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/6_/_7wswsfn4h70ffw10z3vw2vm0000gn/T/pip-build-_gBx7f/cryptography
```
and when I run deken I got
```
deken
/usr/local/bin/deken: line 133: /Users/Pierre/.deken/virtualenv/bin/hy: No such file or directory
```
The problem has already been noticed #168, #169 I think but I don't know how to solve it.

**Comment by umlaeute** [comment]
assuming you are running the latest and greated `deken`, could you please try with `python3` (provided it is installed on your system)?

~~~bash
rm -rf ~/.deken
PYTHON_BIN=python3 ./deken install
~~~

**Comment by pierreguillot** [comment]
I followed the steps in the dev readme, I think I have the lastest version (this is 0.2.5 which is the same on the master branch). Using `PYTHON_BIN=python3`, the installation works but now when I run deken
```
Traceback (most recent call last):
  File "/Users/Pierre/.deken/virtualenv/bin/hy", line 7, in <module>
    from hy.cmdline import hy_main
  File "/Users/Pierre/.deken/virtualenv/lib/python2.7/site-packages/hy/cmdline.py", line 36, in <module>
    import astor.codegen
ImportError: No module named codegen
```

Is it normal that the installation still use python2.7 ?

**Comment by umlaeute** [comment]
i don't think so.

**Comment by pierreguillot** [comment]
Here is the log:
```
Pierre$ PYTHON_BIN=python3 ~/bin/deken install
This is your first time running deken on this machine.
I'm going to install myself and my dependencies into /Users/Pierre/.deken now.
Feel free to ctrl-C now if you don't want to do this.
Fetching Python requirements file: https://raw.githubusercontent.com/pure-data/deken/master/developer/requirements.txt
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   151  100   151    0     0    126      0  0:00:01  0:00:01 --:--:--   184
Fetching main hylang file: https://raw.githubusercontent.com/pure-data/deken/master/developer/deken.hy
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 27063  100 27063    0     0  19908      0  0:00:01  0:00:01 --:--:-- 19943
Downloading & installing Virtualenv.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   122    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   281  100   281    0     0    178      0  0:00:01  0:00:01 --:--:--  1089
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
100 1820k  100 1820k    0     0   164k      0  0:00:11  0:00:11 --:--:--  284k
x virtualenv-15.1.0/
x virtualenv-15.1.0/AUTHORS.txt
x virtualenv-15.1.0/bin/
x virtualenv-15.1.0/bin/rebuild-script.py
x virtualenv-15.1.0/docs/
x virtualenv-15.1.0/docs/changes.rst
x virtualenv-15.1.0/docs/conf.py
x virtualenv-15.1.0/docs/development.rst
x virtualenv-15.1.0/docs/index.rst
x virtualenv-15.1.0/docs/installation.rst
x virtualenv-15.1.0/docs/make.bat
x virtualenv-15.1.0/docs/Makefile
x virtualenv-15.1.0/docs/reference.rst
x virtualenv-15.1.0/docs/userguide.rst
x virtualenv-15.1.0/LICENSE.txt
x virtualenv-15.1.0/MANIFEST.in
x virtualenv-15.1.0/PKG-INFO
x virtualenv-15.1.0/README.rst
x virtualenv-15.1.0/scripts/
x virtualenv-15.1.0/scripts/virtualenv
x virtualenv-15.1.0/setup.cfg
x virtualenv-15.1.0/setup.py
x virtualenv-15.1.0/tests/
x virtualenv-15.1.0/tests/__init__.py
x virtualenv-15.1.0/tests/test_activate.sh
x virtualenv-15.1.0/tests/test_activate_output.expected
x virtualenv-15.1.0/tests/test_cmdline.py
x virtualenv-15.1.0/tests/test_virtualenv.py
x virtualenv-15.1.0/virtualenv.egg-info/
x virtualenv-15.1.0/virtualenv.egg-info/dependency_links.txt
x virtualenv-15.1.0/virtualenv.egg-info/entry_points.txt
x virtualenv-15.1.0/virtualenv.egg-info/not-zip-safe
x virtualenv-15.1.0/virtualenv.egg-info/PKG-INFO
x virtualenv-15.1.0/virtualenv.egg-info/SOURCES.txt
x virtualenv-15.1.0/virtualenv.egg-info/top_level.txt
x virtualenv-15.1.0/virtualenv.py
x virtualenv-15.1.0/virtualenv_embedded/
x virtualenv-15.1.0/virtualenv_embedded/activate.bat
x virtualenv-15.1.0/virtualenv_embedded/activate.csh
x virtualenv-15.1.0/virtualenv_embedded/activate.fish
x virtualenv-15.1.0/virtualenv_embedded/activate.ps1
x virtualenv-15.1.0/virtualenv_embedded/activate.sh
x virtualenv-15.1.0/virtualenv_embedded/activate_this.py
x virtualenv-15.1.0/virtualenv_embedded/deactivate.bat
x virtualenv-15.1.0/virtualenv_embedded/distutils-init.py
x virtualenv-15.1.0/virtualenv_embedded/distutils.cfg
x virtualenv-15.1.0/virtualenv_embedded/python-config
x virtualenv-15.1.0/virtualenv_embedded/site.py
x virtualenv-15.1.0/virtualenv_support/
x virtualenv-15.1.0/virtualenv_support/__init__.py
x virtualenv-15.1.0/virtualenv_support/argparse-1.4.0-py2.py3-none-any.whl
x virtualenv-15.1.0/virtualenv_support/pip-9.0.1-py2.py3-none-any.whl
x virtualenv-15.1.0/virtualenv_support/setuptools-28.8.0-py2.py3-none-any.whl
x virtualenv-15.1.0/virtualenv_support/wheel-0.29.0-py2.py3-none-any.whl
running build
running build_py
creating build
creating build/lib
copying virtualenv.py -> build/lib
creating build/lib/virtualenv_support
copying virtualenv_support/__init__.py -> build/lib/virtualenv_support
copying virtualenv_support/argparse-1.4.0-py2.py3-none-any.whl -> build/lib/virtualenv_support
copying virtualenv_support/pip-9.0.1-py2.py3-none-any.whl -> build/lib/virtualenv_support
copying virtualenv_support/setuptools-28.8.0-py2.py3-none-any.whl -> build/lib/virtualenv_support
copying virtualenv_support/wheel-0.29.0-py2.py3-none-any.whl -> build/lib/virtualenv_support
Setting up the virtual environment.
New python executable in /Users/Pierre/.deken/virtualenv/bin/python
Installing setuptools, pip, wheel...done.
Installing deken library dependencies.
Collecting hy==0.12.1 (from -r /Users/Pierre/.deken/requirements.txt (line 1))
Collecting easywebdav==1.2.0 (from -r /Users/Pierre/.deken/requirements.txt (line 2))
Collecting pyelftools==0.24 (from -r /Users/Pierre/.deken/requirements.txt (line 3))
Collecting macholib==1.8 (from -r /Users/Pierre/.deken/requirements.txt (line 4))
  Using cached macholib-1.8-py2.py3-none-any.whl
Collecting python-gnupg==0.4.1 (from -r /Users/Pierre/.deken/requirements.txt (line 5))
  Using cached python_gnupg-0.4.1-py2.py3-none-any.whl
Collecting keyring==10.4.0 (from -r /Users/Pierre/.deken/requirements.txt (line 6))
  Using cached keyring-10.4.0-py2.py3-none-any.whl
Collecting ndg_httpsclient==0.4.3 (from -r /Users/Pierre/.deken/requirements.txt (line 7))
  Using cached ndg_httpsclient-0.4.3-py2-none-any.whl
Collecting pyasn1==0.1.9 (from -r /Users/Pierre/.deken/requirements.txt (line 8))
  Using cached pyasn1-0.1.9-py2.py3-none-any.whl
Collecting pyOpenSSL==16.2.0 (from -r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached pyOpenSSL-16.2.0-py2.py3-none-any.whl
Collecting clint>=0.4 (from hy==0.12.1->-r /Users/Pierre/.deken/requirements.txt (line 1))
Collecting astor>=0.5 (from hy==0.12.1->-r /Users/Pierre/.deken/requirements.txt (line 1))
  Using cached astor-0.6-py2.py3-none-any.whl
Collecting rply>=0.7.0 (from hy==0.12.1->-r /Users/Pierre/.deken/requirements.txt (line 1))
  Using cached rply-0.7.5-py2.py3-none-any.whl
Collecting requests (from easywebdav==1.2.0->-r /Users/Pierre/.deken/requirements.txt (line 2))
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting altgraph>=0.13 (from macholib==1.8->-r /Users/Pierre/.deken/requirements.txt (line 4))
  Using cached altgraph-0.14-py2.py3-none-any.whl
Collecting cryptography>=1.3.4 (from pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached cryptography-2.1.3-cp27-cp27m-macosx_10_6_intel.whl
Collecting six>=1.5.2 (from pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting args (from clint>=0.4->hy==0.12.1->-r /Users/Pierre/.deken/requirements.txt (line 1))
Collecting appdirs (from rply>=0.7.0->hy==0.12.1->-r /Users/Pierre/.deken/requirements.txt (line 1))
  Using cached appdirs-1.4.3-py2.py3-none-any.whl
Collecting idna<2.7,>=2.5 (from requests->easywebdav==1.2.0->-r /Users/Pierre/.deken/requirements.txt (line 2))
  Using cached idna-2.6-py2.py3-none-any.whl
Collecting urllib3<1.23,>=1.21.1 (from requests->easywebdav==1.2.0->-r /Users/Pierre/.deken/requirements.txt (line 2))
  Using cached urllib3-1.22-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->easywebdav==1.2.0->-r /Users/Pierre/.deken/requirements.txt (line 2))
  Using cached certifi-2017.11.5-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests->easywebdav==1.2.0->-r /Users/Pierre/.deken/requirements.txt (line 2))
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Collecting cffi>=1.7; platform_python_implementation != "PyPy" (from cryptography>=1.3.4->pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached cffi-1.11.2-cp27-cp27m-macosx_10_6_intel.whl
Collecting enum34; python_version < "3" (from cryptography>=1.3.4->pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached enum34-1.1.6-py2-none-any.whl
Collecting asn1crypto>=0.21.0 (from cryptography>=1.3.4->pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached asn1crypto-0.23.0-py2.py3-none-any.whl
Collecting ipaddress; python_version < "3" (from cryptography>=1.3.4->pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
  Using cached ipaddress-1.0.18-py2-none-any.whl
Collecting pycparser (from cffi>=1.7; platform_python_implementation != "PyPy"->cryptography>=1.3.4->pyOpenSSL==16.2.0->-r /Users/Pierre/.deken/requirements.txt (line 9))
Installing collected packages: args, clint, astor, appdirs, rply, hy, idna, urllib3, certifi, chardet, requests, easywebdav, pyelftools, altgraph, macholib, python-gnupg, keyring, pycparser, cffi, enum34, asn1crypto, ipaddress, six, cryptography, pyOpenSSL, pyasn1, ndg-httpsclient
  Installing from a newer Wheel-Version (1.1)
Successfully installed altgraph-0.14 appdirs-1.4.3 args-0.1.0 asn1crypto-0.23.0 astor-0.6 certifi-2017.11.5 cffi-1.11.2 chardet-3.0.4 clint-0.5.1 cryptography-2.1.3 easywebdav-1.2.0 enum34-1.1.6 hy-0.12.1 idna-2.6 ipaddress-1.0.18 keyring-10.4.0 macholib-1.8 ndg-httpsclient-0.4.3 pyOpenSSL-16.2.0 pyasn1-0.1.9 pycparser-2.18 pyelftools-0.24 python-gnupg-0.4.1 requests-2.18.4 rply-0.7.5 six-1.11.0 urllib3-1.22
```

**Comment by pierreguillot** [comment]
I never used python so I can't really tell what's wrong. But perhaps that even if the installation uses python 3, the python version embed in deken seems to be the one offered natively by the system.

**Comment by pierreguillot** [comment]
I managed to solve the problem! I'm sorry, I never use python so perhaps I did a mistake or I have bad practices but I tried on other mac computer and I encountered the same problem. Python seems to be pre-installed on mac (with the version 2.7) so the solution is to install Python3 and to add an [alias to this version](https://stackoverflow.com/questions/18425379/how-to-set-pythons-default-version-to-3-3-on-os-x) before installing Deken. Now everything works perfectly!
